hbase-balancer/src/main/java/org/apache/hadoop/hbase/CatalogAccessor.java

```diff
@@ -0,0 +1,1071 @@
+    scan.withStopRow(stopKey);
+    return scan;
+  }
+
+  private static Scan getCatalogScan(Connection connection, int rowUpperLimit) {
+    Scan scan = new Scan();
+    int scannerCaching = connection.getConfiguration().getInt(HConstants.HBASE_META_SCANNER_CACHING,
+      HConstants.DEFAULT_HBASE_META_SCANNER_CACHING);
+    if (connection.getConfiguration().getBoolean(HConstants.USE_META_REPLICAS,
+      HConstants.DEFAULT_USE_META_REPLICAS)) {
+      scan.setConsistency(Consistency.TIMELINE);
+    }
+    if (rowUpperLimit > 0) {
+      scan.setLimit(rowUpperLimit);
+      scan.setReadType(Scan.ReadType.PREAD);
+    }
+    scan.setCaching(scannerCaching);
+    return scan;
+  }
+
+  /**
+   * Do not use this method to get meta table regions, use methods in RootTableLocator instead.
+   * @param connection connection we're using
+   * @param tableName table we're looking for
+   * @return Return list of regioninfos and server.
+   */
+  public static List<Pair<RegionInfo, ServerName>>
+    getTableRegionsAndLocations(Connection connection, TableName tableName) throws IOException {
+    return getTableRegionsAndLocations(connection, tableName, true);
+  }
+
+  /**
+   * Do not use this method to get meta table regions, use methods in RootTableLocator instead.
+   * @param connection connection we're using
+   * @param tableName table to work with, can be null for getting all regions
+   * @param excludeOfflinedSplitParents don't return split parents
+   * @return Return list of regioninfos and server addresses.
+   */
+  // What happens here when 1M regions in hbase:meta? This won't scale?
+  public static List<Pair<RegionInfo, ServerName>> getTableRegionsAndLocations(
+    Connection connection, @Nullable final TableName tableName,
+    final boolean excludeOfflinedSplitParents) throws IOException {
+    if (tableName != null && tableName.equals(TableName.ROOT_TABLE_NAME)) {
+      throw new IOException(
+        "This method can't be used to locate root regions; use RootTableLocator instead");
+    }
+    // Make a version of CollectingVisitor that collects RegionInfo and ServerAddress
+    ClientMetaTableAccessor.CollectRegionLocationsVisitor visitor =
+      new ClientMetaTableAccessor.CollectRegionLocationsVisitor(excludeOfflinedSplitParents);
+    byte[] startRow = ClientMetaTableAccessor.getTableStartRowForMeta(tableName, QueryType.REGION);
+    byte[] stopRow = ClientMetaTableAccessor.getTableStopRowForMeta(tableName, QueryType.REGION);
+    TableName parentTable = TableName.META_TABLE_NAME;
+    if (TableName.META_TABLE_NAME.equals(tableName)) {
+      parentTable = TableName.ROOT_TABLE_NAME;
+      startRow = null;
+      stopRow = null;
+    }
+
+    scanCatalog(connection,
+      parentTable,
+      startRow,
+      stopRow,
+      QueryType.REGION,
+      Integer.MAX_VALUE,
+      visitor);
+    return visitor.getResults();
+  }
+
+  public static void fullScanMetaAndPrint(Connection connection) throws IOException {
```

Comment:
Just meta table?

Reply:
Yeah keeping legacy apis I don't need to change just as part of this PR to avoid doing pervasive renames.